### DISCUSSION TITLE
CustomBA elevation v5

### DISCUSSION THREAD

**Discussion by ds-andreasor** [wixtoolset/issues#8076] [Questions] [answered]
With the out of process bootstrapper agent it is now possible to add a manifest to the custom BA assembly.
So there are 3 options for early elevation now:
1. Run the bundle elevated i.e. from another program
2. Add a requireAdministrator manifest to the custom BA assembly (not sure when UAC would show up here)
3. Call the `IBoostrapperEngine.Elevate()` first at the run callback
But which is the preferred one?

I know that it shouldn't be done in general but sometimes old software requires doing things like this.

**Comment by barnson** [comment, marked as answer]
A manifest won't elevate a BA. Elevated BAs aren't supported; that hasn't changed.

**Reply by ds-andreasor** [reply]
Okay, got it.

**Comment by robmen** [comment]
Option 3 is part of our security design.

Option 1 exposes more security risk but is handled because it can happen due to user behavior. Half of the security vulnerabilities reported in WiX are due to supporting this case.

Option 2 does not work, and if it was made to work, it would have security problems.

**Reply by ds-andreasor** [reply]
Thx for the clarification Rob.
Right now we use option 1 in the case it can't be avoided.
Will check if option 3 would work and use this if possible.

**Reply by robmen** [reply]
> Right now we use option 1 in the case it can't be avoided.

If you were my customer, we'd work with you to address your installation design issues to not require elevation in your bootstrapper application and improve your security situation.

**Reply by ds-andreasor** [reply]
That is a decision for the company management, but I proposed this already so we will see.
